genie-common-internal/src/main/java/com/netflix/genie/common/internal/dto/v4/JobArchivalDataRequest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.common.internal.dto.v4;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+
+/**
+ * Job archival options for the Genie agent.
+ *
+ * @author standon
+ * @since 4.0.0
+ */
+@Getter
+@ToString(doNotUseGetters = true)
+@EqualsAndHashCode(doNotUseGetters = true)
+@SuppressWarnings("checkstyle:finalclass")
+public class JobArchivalDataRequest {
```

Comment:
I am trying to convey with this name that this is a section of `JobRequest` which includes Job archival data.`JobArchivalRequest` seems like a request to archive the job which is not the right thing in this context

Reply:
By this logic, we should call `JobRequest` `JobRequestData`, no?
Since it's not a request to execute a job (the server executes no job) on that codepath..